### ISSUE TITLE
Unable to connect to the server notification on creating new user accounts

### ISSUE THREAD

**Issue by xelawafs** [KETSE/casebox#68]
I have installed casebox following the [Ubuntu-14.04-x64](https://github.com/KETSE/casebox/wiki/Ubuntu-14.04-x64) installatoin docs with [Apache2 Configuration](https://github.com/KETSE/casebox/wiki/Apache2-configuration).

When I try creating new users, I keep getting this notification on the browser: **Unable to connect to the server.** I've been able to create two users at different occasions. This is after clicking the **save** button several times. But even that's not working anymore.

Any chance this could be a memory issue causing a service to shutdown? I have a 4GB swap file enabled with 512 MBs of RAM

**Comment by vmelnic** [comment]
Hi @xelawafs 
Did you try to increase PHP memory_limit to 256M? 
Also check developer console (firebug) for javascript errors when you submit user form.

Cheers!

**Comment by xelawafs** [comment]
This is what I have on the browser console **right before clicking the save button**:

```
unreachable code after return statement[Learn More] ext-charts.js:1:99805
window.controllers is deprecated. Do not use it for UA detection.  default
Password fields present on an insecure (http://) page. This is a security risk that allows user login credentials to be stolen.[Learn More]  default
Use of getPreventDefault() is deprecated.  Use defaultPrevented instead.
```
No changes on the console when I click save. I've increased PHP's memory_limit to 256M now. I'm going to bump up my RAM to 1GB and see if that makes a difference.

**Comment by oburlaca** [comment]
@xelawafs it's a JS issue, 
I guess you're using Firefox. I found this: http://stackoverflow.com/questions/27368537/firefox-window-controllers-is-deprecated-do-not-use-it-for-ua-detection

not sure it applies to your case

**Comment by xelawafs** [comment]
Yeah, I'm using Firefox. Coincidentally, I just installed LiveHTTPHeaders yesterday :) I've switched to Chrome just to make sure we have all that out of the equation.

I still have the same problem but now we have a useful console log when I click save:

```
POST http://casebox.example.com/c/default/remote/router 500 (Internal Server Error) ext-all.js:22
```

**Comment by xelawafs** [comment]
Tried doing this on a Mac  (Chrome) and it worked fine. Here are Apache access logs for trials on a Mac (Chrome) and Linux (Chrome):

```
[15/Feb/2017:10:51:54 +0000] "POST /c/default/remote/router HTTP/1.1" 200 555 "http://case.example.com/c/default/?action=auth" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKi$
[15/Feb/2017:11:42:13 +0000] "POST /c/default/remote/router HTTP/1.1" 500 743 "http://casebox.example.com/c/default/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko)$
```

I'll try get feedback from a small group of testers to see if I have an isolated case.

**Comment by fapvoice** [comment]
I have tryout to install this on CentOS 7 using https://github.com/KETSE/casebox/wiki/CentOS-7.2-x64 and everything is working fine except user and group creation. When i try to create users and group I am getting _Unable to connect to the server_ error when I try to save it.

**Comment by ronikb** [comment]
I have tryout to install this on CentOS 7 using https://github.com/KETSE/casebox/wiki/CentOS-7.2-x64 and everything is working fine except user and group creation. When i try to create users and group I am getting _Unable to connect to the server_ error when I try to save it.

**Comment by vmelnic** [comment]
Hi @ronikb 
Check the logs from app and/or nginx.

**Comment by vmelnic** [comment]
Hi @ronikb 
Application logs are available in var/logs/ folder (relative to your application docroot. For example /var/www/casebox/var/log/default.log).

**Comment by ronikb** [comment]
Strange... Checked this application log (default.log and db.log) and it seems that nothing logged in this application log file when I get the error. Screenshot of my error attached herewith.
![casebox_error](https://cloud.githubusercontent.com/assets/805095/24101976/53381b36-0da0-11e7-8af1-020f559238f2.JPG)

**Comment by roopeshsivam** [comment]
[2017-03-30 08:20:56] security.DEBUG: Read existing security token from the session. {"key":"_security_main"} []
[2017-03-30 08:20:56] security.DEBUG: User was reloaded from a user provider. {"username":"root","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2017-03-30 08:20:56] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "User Error: 2017-03-30 08:20:56:   <br /><hr />Query error (casebox): <hr /><br />   <br /><hr />Query: INSERT INTO users_groups (`name`,`first_name`,`last_name`,`cid`,`language_id`,`email`,`salt`,`roles`,`cdate`,`type`) VALUES (:p1,:p2,:p3,:p4,:p5,:p6,:p7,:p8,:p9,:p10)2017-03-30 08:20:56:   <br /><hr />Query error (casebox): <hr /><br />  " at /var/www/casebox/vendor/caseboxdev/core-bundle/src/Service/DBService.php line 194 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ContextErrorException(code: 0): User Error: 2017-03-30 08:20:56: \n\r<br /><hr />Query error (casebox): <hr /><br />\n\r\n\r<br /><hr />Query: INSERT INTO users_groups (`name`,`first_name`,`last_name`,`cid`,`language_id`,`email`,`salt`,`roles`,`cdate`,`type`) VALUES (:p1,:p2,:p3,:p4,:p5,:p6,:p7,:p8,:p9,:p10)2017-03-30 08:20:56: \n\r<br /><hr />Query error (casebox): <hr /><br />\n\r at /var/www/casebox/vendor/caseboxdev/core-bundle/src/Service/DBService.php:194)"} []
[2017-03-30 08:20:56] security.DEBUG: Stored the security token in the session. {"key":"_security_main"} []

**Comment by roopeshsivam** [comment]
`[2017-03-30 08:20:56] security.DEBUG: Read existing security token from the session. {"key":"_security_main"} []
[2017-03-30 08:20:56] security.DEBUG: User was reloaded from a user provider. {"username":"root","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2017-03-30 08:20:56] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\ContextErrorException: "User Error: 2017-03-30 08:20:56:   <br /><hr />Query error (casebox): <hr /><br />   <br /><hr />Query: INSERT INTO users_groups (`name`,`first_name`,`last_name`,`cid`,`language_id`,`email`,`salt`,`roles`,`cdate`,`type`) VALUES (:p1,:p2,:p3,:p4,:p5,:p6,:p7,:p8,:p9,:p10)2017-03-30 08:20:56:   <br /><hr />Query error (casebox): <hr /><br />  " at /var/www/casebox/vendor/caseboxdev/core-bundle/src/Service/DBService.php line 194 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\ContextErrorException(code: 0): User Error: 2017-03-30 08:20:56: \n\r<br /><hr />Query error (casebox): <hr /><br />\n\r\n\r<br /><hr />Query: INSERT INTO users_groups (`name`,`first_name`,`last_name`,`cid`,`language_id`,`email`,`salt`,`roles`,`cdate`,`type`) VALUES (:p1,:p2,:p3,:p4,:p5,:p6,:p7,:p8,:p9,:p10)2017-03-30 08:20:56: \n\r<br /><hr />Query error (casebox): <hr /><br />\n\r at /var/www/casebox/vendor/caseboxdev/core-bundle/src/Service/DBService.php:194)"} []
[2017-03-30 08:20:56] security.DEBUG: Stored the security token in the session. {"key":"_security_main"} []
`

**Comment by Pauzotah** [comment]
My default log also shows the same error as roopeshsivam.

Has anyone found the problem, everything appears to run fine otherwise!

**Comment by colindoyle** [comment]
I am having the same issue.  I attempt to set the Can add User: check box under Account and then save it.  When I go back into accounts the box is unchecked.
I suspect that it has something to do with the initial setup of the root account.
